test/src/main/java/org/apache/accumulo/test/start/KeywordStartIT.java

```diff
@@ -203,10 +208,41 @@ public void checkHasMain() {
     expectSet.add(TabletServer.class);
     expectSet.add(ZooKeeperMain.class);
 
+    expectSet.add(ConvertConfig.class);
+    expectSet.add(ConfigPropertyUpgrader.class);
+    expectSet.add(CheckServerConfig.class);
+    expectSet.add(CreateEmpty.class);
+    expectSet.add(ECAdmin.class);
+    expectSet.add(GenerateSplits.class);
+    expectSet.add(SplitLarge.class);
+    expectSet.add(ZooZap.class);
+
+    // check that classes in the expected set contain a main
     for (Class<?> c : expectSet) {
       assertTrue(hasMain(c), "Class " + c.getName() + " is missing a main method!");
     }
 
+    // build a list of all classed that implement KeywordExecutable
+    TreeMap<String,KeywordExecutable> foundExecutables =
+        new TreeMap<>(Main.getExecutables(getClass().getClassLoader()));
```

Comment:
This only checks for KeywordExecutables that have a main method that we didn't expect. My comment was to check all classes, not just KeywordExecutable classes. So, this is a start, but incomplete (relative to the suggestion).